### PR TITLE
Adding pulse tests

### PR DESCRIPTION
**Pull request by georgesbarron**
### Summary

Adding tests in `qiskit.pulse` for
- `Interval`
- `TimeSlot`
- `LoRange`
- `LoConfig`

Added error raised for given beginning time being after ending time in `Interval`.

### CODE REVIEW

File: test/python/pulse/test_experiment_configurations.py [hunk 1]
```diff
@@ -17,10 +17,30 @@
...
+        """Test creation of LoRange. Test upper/lower bounds and includes.
+        Test __eq__ for two same and different LoRange's.
+        """
+        lo_range_1 = LoRange(lower_bound=-0.1, upper_bound=+0.1)
+
+        self.assertEqual(lo_range_1.lower_bound, -0.1)
+        self.assertEqual(lo_range_1.upper_bound, +0.1)
+        self.assertTrue(lo_range_1.includes(0.0))
+
+        lo_range_2 = LoRange(upper_bound=+0.1, lower_bound=-0.1)
```
Comment (taalexander) [inline test/python/pulse/test_experiment_configurations.py]: Could you please reverse the order of these kwargs?

File: test/python/pulse/test_timeslots.py [hunk 2]
```diff
@@ -64,6 +87,18 @@ def test_default(self):
...
+        """Test shifting of Timeslot."""
+        slot_1 = Timeslot(Interval(1, 3), AcquireChannel(0))
+        slot_2 = Timeslot(Interval(1+5, 3+5), AcquireChannel(0))
+        shifted_slot = slot_1.shift(+5)
+        self.assertTrue(slot_2 == shifted_slot)
+
+    def test_representation_of_timeslot_object(self):
+        """Test representation of Timeslot object."""
+        slot = Timeslot(Interval(1, 5), AcquireChannel(0))
+        self.assertEqual(str(slot), 'Timeslot(AcquireChannel(0), (1, 5))')
```
Comment (taalexander) [inline test/python/pulse/test_timeslots.py]: Could you make this a call to `repr` please and thank you 😄.

File: test/python/pulse/test_experiment_configurations.py (hunk 1 above)
Comment (georgesbarron) [reply test/python/pulse/test_experiment_configurations.py]: Done! https://github.com/Qiskit/qiskit-terra/pull/2730/commits/ee3063dea8faa64e09aeeecba314dfba468e6a2d

File: test/python/pulse/test_timeslots.py (hunk 2 above)
Comment (georgesbarron) [reply test/python/pulse/test_timeslots.py]: Done! https://github.com/Qiskit/qiskit-terra/pull/2730/commits/4d746014157b7659aecf9bbfe7007082e3c1a530

**Pull request merged**